src/util/system.cpp

```diff
@@ -784,7 +789,8 @@ void ClearDatadirCache()
 
 fs::path GetConfigFile(const std::string& confPath)
 {
-    return AbsPathForConfigVal(fs::path(confPath), false);
+    fs::path path = fs::path(confPath);
+    return path.is_absolute() ? path : AbsPathForConfigVal(path, false);
```

Comment:
Is this necessary? Looks like `fs::absolute` already does this in `AbsPathForConfigVal`.

Reply:
Yes, it is needed to not touch default datadir unnecessarily, before all configs are read.

Reply:
It seems like a better fix (simpler and more general) would be to add:

```c++
if (path.is_absolute()) return path;
```

as the first line of AbsPathForConfigVal.